ceph/ceph_admin/add.py

```diff
@@ -0,0 +1,48 @@
+from typing import Dict, Union
+
+from ceph.utils import get_node_by_id
+
+from .common import config_dict_to_string
+from .typing_ import DaemonProtocol, OrchProtocol
+
+
+class AddMixin:
+    def add(self: Union[OrchProtocol, DaemonProtocol], config: Dict):
+        """
+
+
+        Execute the add method using the object's service name.
+        Args:
+            config:     Key/value pairs passed from the test suite.
+                        pos_args        - List to be added as positional params
+
+        Example:
+            config:
+              service: osd
+              command: add
+              base_cmd_args:
+                verbose: true
+              pos_orgs:
+                - node1
+                - /dev/vdb
+        """
+        service = config.pop("service")
+        base_cmd = ["ceph", "orch", "daemon", "add", service]
+        if config.get("base_cmd_args"):
+            base_cmd.append(config_dict_to_string(config["base_cmd_args"]))
+
+        pos_args = config["pos_args"]
+        node = pos_args[0]
+        host_id = get_node_by_id(self.cluster, node)
+        host = host_id.shortname
+
+        if service == "osd":
+            base_cmd.extend([f"{host}:{pos_args[1]}"])
```

Comment:
IMO, I think it would create a problem if multiple devices are passed from test case config.
ex.,   
```
pos_args:
   - node1
   - /dev/vdb
   - /dev/vdc
```

Reply:
you might have to use `join` statement to join all devices of a node, which is later added as OSD.

Reply:
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-1626968379649/Add_osd_on_single_device_0.log